ir/src/transition_constraints/graph.rs

```diff
@@ -187,9 +195,69 @@ impl AlgebraicGraph {
                 let node_index = self.insert_op(Operation::PeriodicColumn(index, cycle_len));
                 Ok((trace_segment, node_index))
             }
+            IdentifierType::Constant(ConstantType::Scalar(scalar_const)) => {
+                let trace_segment = 0;
+                let node_index = self.insert_op(Operation::ScalarConstant(scalar_const));
+                Ok((trace_segment, node_index))
+            }
             _ => Err(SemanticError::InvalidUsage(format!(
-                "Identifier {} was declared as a {} not as a trace column",
-                ident, col_type
+                "Identifier {} was declared as a {} which is not a supported type.",
+                ident, elem_type
+            ))),
+        }
+    }
+
+    /// Validates and adds a vector access to the graph.
+    /// Returns an error if the vector identifier's value is not of a supported type.'
+    fn insert_vector_access(
+        &mut self,
+        symbol_table: &SymbolTable,
+        vector_access: &VectorAccess,
+    ) -> Result<(TraceSegment, NodeIndex), SemanticError> {
+        // check that vector access is valid
+        symbol_table.validate_vector_access(vector_access.name(), vector_access.idx())?;
+        let vector_access_type = symbol_table.get_type(vector_access.name())?;
```

Comment:
Thanks. I changed it to pass `access` struct directly but I think keeping it as `validate_vector_access()` seems better. I moved the symbol table lookup above the validation check and pass the `IdentifierType` to the `validate_vector_access()` method so that we have to do the lookup only once. Using `access_vector_element()` in cases where we ignore the value was looking a little weird. Please let me know if you think otherwise and would prefer `access_vector_element()` instead.

Reply:
This could work but then I don't think `SymbolTable` is the right place for these functions since we are not using `self` inside these functions at all. If you want to keep it this way, I'd move the function somewhere else - e.g., either as a helper function in `graph.rs` or maybe as method on `VectorAccess` and `MatrixAccess`.

Reply:
Would it make sense to keep the functions as public helper functions in `symbol_table.rs` file itself since we use them in `graph.rs` and `boundary_constraints.rs` and we already have other helper validation functions in `symbol_table.rs`?